src/test/java/core/basesyntax/dao/FileReaderImplTest.java

```diff
@@ -0,0 +1,61 @@
+package core.basesyntax.dao;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class FileReaderImplTest {
+    private static final String INPUT_FILE = "input.csv";
+    private static final String EMPTY_FILE = "empty.csv";
+    private static final String INCORRECT_FILE = "incorrect.csv";
+    private static final String NON_EXISTENT_FILE = "nonexistent.csv";
+    private static final String CORRECT_FILE = "src/test/resources/transactions.csv";
+    private static FileReaderImpl fileReader;
+
+    @BeforeEach
+    void setUp() {
+        fileReader = new FileReaderImpl();
+    }
+
+    @Test
+    void read_Empty_File_notOk() {
+        assertThrows(RuntimeException.class,
+                () -> fileReader.read(EMPTY_FILE));
+    }
+
+    @Test
+    void read_incorrectFile_notOk() {
+        assertThrows(RuntimeException.class,
+                () -> fileReader.read(INCORRECT_FILE));
+    }
+
+    @Test
+    void read_correctFile_ok() {
+        assertThrows(RuntimeException.class,
+                () -> fileReader.read(INPUT_FILE));
+    }
+
+    @Test
+    public void read_emptyFile_ok() {
+        assertThrows(RuntimeException.class,
+                () -> fileReader.read(EMPTY_FILE));
+    }
```

Comment:
you have a lot of test methods out here but most of them are wrong and redundant. For example, is this file really empty? I think the file doesn't exist at all. And what is the difference between INCORRECT_FILE and NON_EXISTENT_FILE ? Please recheck all your test methods in all classes and make your solution clear and concise